### PR TITLE
Context menu changes

### PR DESCRIPTION
**Pull request by royderks**
## Amsterdam styled components pull request

Before opening a pull request, please ensure:

- [ ] The default export from a file matches the file name
- [ ] The component styles are only placed in the `asc-core/src/components` folder. Exception is the `asc-ui/src/internals` folder where the styles are allowed. The components from this folder are just for internal use in the stories.
- [ ] The component style name follows the pattern `<ComponentName>Style/<ComponentName>Style.ts`
- [ ] Pull request has tests (we are going for 100% coverage!)

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx [hunk 1]
```diff
@@ -5,24 +5,43 @@ import MenuList from './ContextMenuList'
...
 
+  componentDidUpdate(prevProps: Props) {
+    const { open } = this.props;
+
+    if (prevProps.open !== open) {
+      this.setState({ open })
+    }
+  }
+
+  list = React.createRef<HTMLDivElement>()
```
Comment (aditudorache) [inline packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx]: Member name is unclear, for list and root, what are they for? 
Please move the member declarations to the begining of the class definition

File: packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx [hunk 2]
```diff
@@ -5,24 +5,43 @@ import MenuList from './ContextMenuList'
...
+      open: props.open,
+      selectedChild: selectedChildInitial,
+    }
+  }
 
+  componentDidUpdate(prevProps: Props) {
+    const { open } = this.props;
+
+    if (prevProps.open !== open) {
+      this.setState({ open })
```
Comment (aditudorache) [inline packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx]: I get this linting error:
Do not use setState in componentDidUpdateeslint(react/no-did-update-set-state)

File: packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx (hunk 1 above)
Comment (aditudorache) [reply packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx]: LInting warning: Enforce component methods order (react/sort-comp)

Comment (royderks) [reply packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx]: Agree, replaced `root` with `wrapper`. So `wrapper` is for `ContextMenuWrapper` and `list` for `ContextMenuList`

File: packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx (hunk 2 above)
Comment (royderks) [reply packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx]: Replaced with a function

**Pull request merged**